rise/static/main.js

```diff
@@ -13,82 +13,104 @@ define([
   'jquery',
   'base/js/namespace',
   'base/js/utils',
-  'services/config',
-], function(require, $, Jupyter, utils, configmod) {
+], function(require, $, Jupyter, utils) {
+
+  "use strict";
 
   /*
-   * Add customized config on top of the default options using the notebook metadata
-   * or the config-derived values
+   * load configuration
+   *
+   * 1) start from the hardwired settings (in this file)
+   * 2) add the settings from nbextensions_configurator (i.e. .jupyter/nbconfig/notebook.json)
+   *    they should all belong in the 'rise' category
+   * 3) and finally add the settings from the notebook metadata
+   *   3a) for legacy reasons: use the 'livereveal' key
+   *   3b) for more consistency, then override with the 'rise' key
+   * 4) also stores in complete_config.is_slideshow a boolean
+   *    that says if the notebook metadata contains 
+   *    either livereveal or rise
+   * 
```

Comment:
What about the config coming from `livereveal.json`?

That is loaded here in https://github.com/damianavila/RISE/pull/344/files#diff-baac928f853273e00536869aeabb2e89L64

IMO, that should be the second step... if we do not add it somehow here we are probably breaking people using that config file to load their config options.

Reply:
Oops, I feel like I missed something big here.  
When reading the code at first I thought that the 'livereveal' ConfigSection corresponded to the 'livereveal' area in the metadata;  
I have never seen nor managed a file named `livereveal.json`  
Can you please elaborate on how that is working in the current official release, so that I can adapt this code to behave in a backward-compatible way.

Reply:
> When reading the code at first I thought that the 'livereveal' ConfigSection corresponded to the 'livereveal' area in the metadata;

Well that's not the case ;-)

> Can you please elaborate on how that is working in the current official release, so that I can adapt this code to behave in a backward-compatible way.

If you use python to customize your config as described here: https://damianavila.github.io/RISE/customize.html#using-python

You are actually creating a `livereveal.json` file which contains the values you customized and it is read by RISE in the line I linked above.

You can try running that Python code I reference above and you will see a new `livereveal.json` file at PREFIX/etc/jupyter/nbconfig.

For instance,
```bash
 cat /Users/davila/miniconda3/envs/rise_stable/etc/jupyter/nbconfig/livereveal.json
{
  "theme": "sky",
  "transition": "zoom",
  "start_slideshow_at": "selected"
}
```

Reply:
OK, I think I now have a more accurate mental model. 

It means in particular that the code as it is right now, as you pointed out in your first review, is not going to work properly wrt the json files. I need to find the time to fix that.

Reply:
>It means in particular that the code as it is right now, as you pointed out in your first review, is not going to work properly wrt the json files

Yep.

> I need to find the time to fix that.

No problem, whenever you can!